runtime/near-vm-runner/src/errors.rs

```diff
@@ -1,5 +1,24 @@
+use core::fmt;
 use near_vm_errors::VMError;
 
 pub trait IntoVMError {
     fn into_vm_error(self) -> VMError;
 }
+
+#[derive(Debug, PartialEq)]
+pub struct ContractPrecompilatonError {
+    // TODO: actually make this a distinct error
+    inner: VMError,
```

Comment:
What is the reason of design error this way instead of make it as a variant of VMError?

Reply:
Yeah, that's what todo is about, it's just  we don't want to conflict with error rework PR here.

Reply:
Not sure why we shall ever make it variant of VMError, guess we uniformly rely on `VMError` for most of error passing in runtime. But I'm open to suggestions, what could be improved here.